File: c/tree/path-sum.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
bool hasPathSum(struct TreeNode* root, int sum) {
    if(root == NULL)
        return false;

    if(root->left == NULL && root->right == NULL)
        return root->val == sum;

    bool bleft = hasPathSum(root->left,sum - root->val);
    if(bleft)
        return true;

    bool bright = hasPathSum(root->right, sum - root->val);

    return bright;
}

File: python/tree/path-sum.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def hasPathSum(self, root, sum):
        if root is None:
            return False

        if root.left is None and root.right is None:
            return root.val == sum

        bleft = self.hasPathSum(root.left, sum - root.val)

        return bleft if bleft else self.hasPathSum(root.right, sum - root.val)
